store/tikv/client.go

```diff
@@ -129,8 +178,16 @@ func (c *batchCommandsClient) batchRecvLoop(cfg config.TiKVClient) {
 	}()
 
 	for {
-		// When `conn.Close()` is called, `client.Recv()` will return an error.
-		resp, err := c.client.Recv()
+		var err error = nil
+		var resp *tikvpb.BatchCommandsResponse = nil
+		failpoint.Inject("gotErrorInRecvLoop", func(_ failpoint.Value) {
+			err = errors.New("injected error in batchRecvLoop")
+		})
+		if err == nil {
+			// When `conn.Close()` is called, `client.Recv()` will return an error.
+			resp, err = c.client.Recv()
+		}
+
 		if err != nil {
```

Comment:
we need refine this error check in next, for error like "over max limit" should not recreate stream?

Reply:
I'm not sure, we can test it later.